ArcadiaTeamsBot.ServiceDesk.Requests/RequestTypeFactory/RequestTypeForUIFactory.cs

```diff
@@ -0,0 +1,41 @@
+﻿namespace ArcadiaTeamsBot.ServiceDesk.Requests.RequestTypeFactory
+{
+    using System;
+    using System.Collections.Generic;
+
+    using ArcadiaTeamsBot.ServiceDesk.Abstractions.DTOs;
+    using ArcadiaTeamsBot.ServiceDesk.Requests.RequestType;
+
+    public class RequestTypeForUIFactory : IRequestTypeForUIFactory
+    {
+        private readonly RequestTypesMappingConfiguration requestTypesMappingConfiguration;
+
+        public RequestTypeForUIFactory(RequestTypesMappingConfiguration requestTypesMappingConfiguration)
+        {
+            this.requestTypesMappingConfiguration = requestTypesMappingConfiguration;
+        }
+
+        public IRequestTypeForUI CreateRequestTypeForUI(ServiceDeskRequestTypeDTO requestTypeDTO)
+        {
+            var requestTypeForUIFields = new List<RequestTypeForUIField>();
+
+            this.requestTypesMappingConfiguration.TryGetValue(requestTypeDTO.Title, out var requestTypeFromConfig);
+
+            foreach (var fieldFromDTO in requestTypeDTO.RequestTypeFields)
+            {
+                var fieldName = fieldFromDTO.FieldName;
+
+                if (requestTypeFromConfig == null ||
```

Comment:
Use null-propagation operator and get rid of Enum.TryParse by changing settings model.

Reply:
I've changed settings model and condition structure.